runtime/src/identity.rs

```diff
@@ -555,6 +561,53 @@ decl_module! {
             }
         }
 
+        // Manage generic authorizations
+        /// Adds an authorization
+        pub fn add_authorization(
+            origin,
+            target_did: IdentityId,
+            authorization_data: AuthorizationData,
+            expiry: Option<T::Moment>
+        ) -> Result {
+            let sender_key = Key::try_from(ensure_signed(origin)?.encode())?;
+            let from_did =  match Self::current_did() {
+                Some(x) => x,
+                None => {
+                    if let Some(did) = Self::get_identity(&sender_key) {
+                        did
+                    } else {
+                        return Err("did not found");
+                    }
+                }
+            };
+
+            Self::add_auth(from_did, target_did, authorization_data, expiry);
+
+            Ok(())
+        }
+
+        /// Removes an authorization
+        pub fn remove_authorization(
+            origin,
+            target_did: IdentityId,
+            auth_id: u64
+        ) -> Result {
+            let sender_key = Key::try_from(ensure_signed(origin)?.encode())?;
```

Comment:
Duplication of code can be avoided if you can make a public helper function that can also be called by the other modules to validate this.

Reply:
Yeah, there's already a code hygiene task around this. Currently, I'm not even sure if this is the best way to do it. If we decide to force identities on multi_sigs, this can be simplified.